javascript/javascript1/week4/voice-assistant.js

```diff
@@ -0,0 +1,219 @@
+                        console.log('No name exist !');
+                    }
+                    else
+                    {
+                        console.log(`your name is: ${nameArray}`)  
+                    }
+                }
+                else if(nameString2 == 'is on myToDo?')
+                {
+                   console.log(`ToDo is: ${todoArray}`);  
+                }
+                else if(nameString2 == 'day is today?')
+                {
+            
+                    const date = new Date().toDateString().split(" ");
+                    console.log(`${date[2]} of ${date[1]} ${date[3]}`);
+                }
+                else
+                {
+                    // mathematic operations
+                    let operator = wordString.slice(-2,-1);  // get the maths operator
+                    let operandA = parseInt(wordString.slice(-3,-2));
+                    let operandB = parseInt(wordString.slice(-1));
+                    let result;
+                    switch (operator[0])
+                    {
+                        case '+' :
+                        result  = operandA + operandB;
+                        console.log(`Result is: ${result}`);  
+                        break;
+
+                        case '-' :
+                        result  = operandA - operandB;
+                        console.log(`Result is: ${result}`);  
+                        break;
+                        
+                        case '*' :
+                        result  = operandA * operandB;
+                        console.log(`Result is: ${result}`);  
+                        break;
+                         
+                        case '/' :
+                        result  = operandA / operandB;
+                        console.log(`Result is: ${result}`);  
+                        break;
+
+                        case '%' :
+                        result  = operandA % operandB;
+                        console.log(`Result is: ${result}`);  
+                        break;
+
+                        default:
+                        console.log('Invalid command !');  
+                        break;  
+                    }
+                }           
+                break;
+
+            //-----------------------------------------
+            case 'Set':  
+                nameString2 = wordString.slice(1, 3); // get keyword like 'a timer'
+                nameString2 = nameString2.join(' '); // combine element to form the string
+                
+                if(nameString2 == 'a timer')
+                {
+                    let units = wordString.slice(-1); 
+                    let timervalue = wordString.slice(-2,-1);  
+                    console.log(`Timer set for ${timervalue} ${units}`);  
+                    console.log('Waiting for time out ....'); 
+                    
+                    let timeout;
+                    if((units='seconds') || (units='sec') )
```

Comment:
This won't work.
If you want to check a value you need to use at least `==`. Ideally `===`.